### PR TITLE
HTTPCLIENT-2065 - Simplify addition of content type parameters in MultipartEntityBuilder

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/MultipartEntityBuilder.java [hunk 1]
```diff
@@ -112,6 +112,17 @@ public MultipartEntityBuilder setContentType(final ContentType contentType) {
...
         return this;
     }
+    /**
+     *  Add parameter to the current {@link ContentType}.
+     *
+     * @param parameter The name-value pair parameter to add to the {@link ContentType}.
+     * @return the {@link MultipartEntityBuilder} instance.
+     * @since 5.2
+     */
+    public MultipartEntityBuilder addParameter(final BasicNameValuePair parameter){
```
Comment (michael-o) [inline httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/MultipartEntityBuilder.java:122]: Space missing: `){`

**Review by michael-o — CHANGES_REQUESTED** [review]

File: httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/MultipartEntityBuilder.java (hunk 1 above)
Comment (arturobernalg) [reply httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/MultipartEntityBuilder.java]: Yep. Fix it.
TY

File: httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java [hunk 2]
```diff
@@ -118,6 +118,20 @@ public void testMultipartCustomContentTypeParameterOverrides() throws Exception
...
+    public void testMultipartCustomContentTypeUsingAddParameter() {
+        final MultipartEntityBuilder eb = MultipartEntityBuilder.create();
+        eb.setMimeSubtype("related");
+        eb.addParameter(new BasicNameValuePair("boundary", "yada-yada"));
+        eb.addParameter(new BasicNameValuePair("charset", "ascii"));
+        eb.addParameter(new BasicNameValuePair("my", "stuff"));
+        eb.buildEntity();
+        final MultipartFormEntity entity =  eb.buildEntity();
+        Assert.assertNotNull(entity);
+        Assert.assertEquals("multipart/related; boundary=yada-yada; charset=US-ASCII; my=stuff",
```
Comment (michael-o) [inline httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java:131]: Interesting, you have added `ascii`, but the value has been changed. Is there any special handling? This is maybe related to https://issues.apache.org/jira/browse/HTTPCLIENT-2159.

Comment (arturobernalg) [reply httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java:131]: yes, looks weird. I gonna take a look to the ticket HTTPCLIENT-2159.

Comment (ok2c) [reply httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java:131]: > Interesting, you have added `ascii`, but the value has been changed. 

@michael-o I think the charset value got correctly normalized to `US-ASCII`.

**Comment by thc202** [comment]
The PR title (and JIRA issue) has a typo in addition.

**Comment by arturobernalg** [comment]
> The PR title (and JIRA issue) has a typo in addition.

TY, Fixed

File: httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java (hunk 2 above)
Comment (michael-o) [reply httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java:131]: True...

**Review by ok2c — APPROVED** [review]

**Review by michael-o — APPROVED** [review]

**Review by michael-o — APPROVED** [review]

**Pull request closed without merging**